server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -843,6 +845,26 @@ List<String> getPipelineStack() {
         return pipelineStack;
     }
 
+    /**
+     * @return Whether a self referencing check should be performed
+     */
+    public boolean isDoNoSelfReferencesCheck() {
+        return doNoSelfReferencesCheck;
+    }
+
+    /**
+     * Whether the ingest framework should perform a self referencing check after this ingest document
+     * has been processed by all pipelines. Doing this check adds an extra tax to ingest and should
+     * only be performed when really needed. Only if a processor is executed that could add self referencing
+     * maps or lists then this check must be performed. Most processors will not be able to do this, hence
+     * the default is <code>false</code>.
+     *
+     * @param doNoSelfReferencesCheck Whether a self referencing check should be performed
+     */
+    public void setDoNoSelfReferencesCheck(boolean doNoSelfReferencesCheck) {
```

Comment:
I think we don't need to conform to the name of the original check (which makes this quite a mouthful). Perhaps we could call this `doSelfReferenceCheck(boolean doCheck)` so it is a little easier to tell what it does from the name?